Add .env file, update issue processing, and include requirements.txt

joyael opened the pull request:
This pull request introduces several key enhancements:

1. Configured `.env` file for secure storage of GitHub and Google Sheets credentials.
2. Updated issue fields and spreadsheet write logic to improve data processing.
3. Added feature to retrieve full name from GitHub username.
4. Introduced `requirements.txt` for dependency management.

joyael committed:
Configured .env for GitHub & Google Sheets credentials, updated issue fields & spreadsheet write logic, added GitHub username to full name resolution, and added requirements.txt for dependency management

AbhijithHaridas commented on ISSUE_SYNC/issue_sync_script.py:
Move this to main.py at the root

AbhijithHaridas commented on .gitignore:
Create a dir called config and gitignore that

AbhijithHaridas replied:
Move env to it as well

AbhijithHaridas reviewed (changes requested):
@joyael

joyael committed:
Refactored project structure: removed ISSUE_SYNC dir, moved script to main.py, and reorganized config folder with ignored credentials

joyael committed:
feat: add authenticated GitHub user fetch, collaborators API, status validation, and color formatting

- Made GitHub user full name fetch use authenticated request
- Added function to get repo collaborators using API
- Implemented color formatting for dropdown options
- Restructured code into function-based structure
- Added data validation logic for status options

joyael committed:
Removed colour styling feature

joyael committed:
Made code cleanup and fixes:
- Changed 'scopes' to a list for Google API.
- Removed unused 'dropdown_values'.
- Simplified set creation from body data.
- Added comment for role placeholder '_'.
- Skipped issues with status 'Backlog'.
- Fixed wrong 'issues_count'.

joyael committed:
Refactor issue sheet logic: add env update, merging functions, and new month handling

- Added function to clear cell values for specific range
- Implemented unmerge and merge functions for sheets
- Added support for monthly sheet duplication and data transfer
- Handled issues with no assignees
- Reorganized code into separate files
- Excluded previous month's issues from current sheet
- Added .env variable update and dynamic range fetching

joyael committed:
Update sheet_functions.py

Added error checking if already the sheet of current month exists.

joyaeljose-yatnam committed:
Full issues fetching

Issues fetching full count instead of limited number of issues.

joyael committed:
Create sonar-project.properties

created file for sonarqube project

joyaeljose-yatnam committed:
corrected comment indentation

joyaeljose-yatnam committed:
Added mode of new spreadsheet and inclusion of only the valid project statuses

joyaeljose-yatnam committed:
Added role_values and assignee_names, output closed issues instead of 'Done' issues, corrected range for clearing sheet values, removed code data validation, collected previous issues, and added output details printing

joyaeljose-yatnam committed:
removed data validation setting functions, removed unwanted variables , removed endrowindex on cleraing cells to clear all rows,  set only one sorting,

joyaeljose-yatnam committed:
closed date based sorting and clearing sheet range modification

joyaeljose-yatnam committed:
ignored __pycache__ folder

joyaeljose-yatnam committed:
Debug printing of 'Done' but not 'closed' issues

joyaeljose-yatnam committed:
gitignore __pycache__ folder

joyaeljose-yatnam committed:
deleted __pycache__ folder

joyaeljose-yatnam committed:
Refactor project status handling and update issue extraction logic
